### PR TITLE
Split PHP milestones into smaller milestones

### PR DESCRIPTION
**Pull request by Potherca**
As discussed during last week's meeting, this MR splits various larger milestones into smaller ones.

Most notably, for the first milestone "Identity" the read-only profile has been separated from the Authentication milestone.

Later milestones have the generic PHP part split off from the Nextcloud integration.

To make the distinction between larger and smaller milestones clear, I have also added an overview at the beginning.

I have tried to leave the original wording intact as much as I could.

### CODE REVIEW

File: milestones.md [hunk 1]
```diff
@@ -1,90 +1,261 @@
...
+of 50,000 euros.
+
+The milestones are:
+
+1. Identity
+   - 1a. Test Suite (€5,000)
+   - 1b. Standalone PHP (€2,500)
+   - 1c. Nextcloud Integration (€2,500)
+   - 1d. Standalone PHP server(€2,500)
+   - 1c. Nextcloud Integration (€2,500)
```
Comment (michielbdejong) [inline milestones.md:16]: duplicate of line 14?

File: milestones.md [hunk 2]
```diff
@@ -1,90 +1,261 @@
...
+   - 2a. Test Suite (€5,000)
+   - 2b. Standalone PHP (€2,500)
+   - 2c. Nextcloud Integration (€2,500)
+
+3. Web ACL
+   - 3a. Test Suite (€5,000)
+   - 3b. Standalone PHP (€2,500)
+   - 3c. Nextcloud Integration (€2,500)
+
+4. Deep Integration
```
Comment (michielbdejong) [inline milestones.md:28]: deep integration only applies to Nextcloud, we integrate *with* Nextcloud. The stand-alone version is already complete at this point, it has nothing to integrate with. :)

File: milestones.md [hunk 3]
```diff
@@ -1,90 +1,261 @@
...
+provider.
+
+### 1a. Test Suite
+
+This milestone is for writing the parts of the test-suite that relate to:
+
+- Correctness of the authorization endpoint
+- Fetching the OpenID config
+- Fetching the WebID profile
+- Working with web tokens
```
Comment (michielbdejong) [inline milestones.md:65]: I would call this part "Generating access tokens"

File: milestones.md [hunk 4]
```diff
@@ -1,90 +1,261 @@
...
+The exact test criteria this test suite tests for will be detailed on a
+dedicated public web page on https://pdsinterop.org/, with links to the relevant
+sections in the official Solid spec.
+
+### 1b. Standalone PHP
+
+This milestone will be for creating an empty standalone PHP server (that all
+following standalone PHP milestones will be implemented in) and an identity
+(WebID profile) that passes the relevant tests from the Identity Test Suite
+milestone.
```
Comment (michielbdejong) [inline milestones.md:79]: I think we should keep 1b and 1d together, see Slack

File: milestones.md [hunk 5]
```diff
@@ -1,90 +1,261 @@
...
+
+### 1c. Nextcloud Integration
```
Comment (michielbdejong) [inline milestones.md:81]: call this something different from 1e, e.g. "Nextcloud read-only profile"

File: milestones.md [hunk 6]
```diff
@@ -1,90 +1,261 @@
...
+
+This milestone will be for implementing a Solid identity provider in PHP
+
+Any generic parts will be implemented as package(s) that can be re-used by the
+Nextcloud integration (and any future PHP implementations).
+
+The implementation will pass rest of the test in the test-suite from the
+Identity Test Suite milestone.
+
+### 1e. Nextcloud Integration
```
Comment (michielbdejong) [inline milestones.md:99]: Call this something different from 1c, e.g. "Nextcloud Identity Provider"

File: milestones.md [hunk 7]
```diff
@@ -1,90 +1,261 @@
...
+
+This milestone will include the integration of the Web Access Control spec into
+the Nextcloud app (using any packages created in the previous milestone).
+
+The implementation will pass all relevant tests from the Web ACL Test Suite
+milestone.
+
+## 4. Deep Integration
+
+### Solid App Launcher
```
Comment (michielbdejong) [inline milestones.md:192]: Keep this as 4a, Nextcloud only

File: milestones.md [hunk 8]
```diff
@@ -1,90 +1,261 @@
...
+This milestone will implement the Solid App Launcher in standalone PHP, in such
+a way that any generic parts can be re-used (as package) in the Nextcloud
+integration.
+
+### 4b. Nextcloud Integration
+
+This milestone will include the integration of the Solid App Launcher into the
+Nextcloud app (using any packages created in the previous milestone).
+
+### Contacts and Profile
```
Comment (michielbdejong) [inline milestones.md:214]: Keep this as 4b, Nextcloud only. Mention that it's in addition to the "Nextcloud read-only profile" milestone

File: milestones.md (hunk 2 above)
Comment (Potherca) [reply milestones.md:28]: Ah, check. Will remove the standalone split.

File: milestones.md (hunk 4 above)
Comment (Potherca) [reply milestones.md:79]: Agreed, will combine them.

File: milestones.md (hunk 6 above)
Comment (Potherca) [reply milestones.md]: Is that still needed with the split items combined again?

File: milestones.md (hunk 5 above)
Comment (Potherca) [reply milestones.md]: Is that still needed with the split items combined again?

**Comment by Potherca** [comment]
@michielbdejong I've implemented the changes as discussed. Could you review and your leasure?

File: milestones.md (hunk 5 above)
Comment (michielbdejong) [reply milestones.md]: No, I think it's fine the way it is in master? https://github.com/pdsinterop/project-admin/blob/master/milestones.md#1c-nextcloud-integration

File: milestones.md (hunk 6 above)
Comment (michielbdejong) [reply milestones.md]: No, I think it's fine the way it is in master?

**Comment by michielbdejong** [comment]
The diff at https://github.com/pdsinterop/project-admin/pull/19/files is quite hard to read; what does this PR do now? Just split milestone 2b into two parts, one for stand-alone and one for Nextcloud-app?

**Comment by Potherca** [comment]
It splits the Storage API milestone `2b. Implementation (€5,000)` into `2b. Standalone PHP (€2,500)` and `2c. Nextcloud Integration (€2,500)` and the Web ACL milestone `3b. Implementation (€5,000)` into `3b. Standalone PHP (€2,500)` and `3c. Nextcloud Integration (€2,500)`

Other changes are mostly structural in nature:

- Adding a summary
- Moving generic explanations from a sub-section into the main section of a milestone
- Spreading "dense" blocks of text across several lines

The only textual changes are explicitly mentioning the intent to create and (re-)use PHP packages to support the overall PHP landscape/community.

File: milestones.md (hunk 6 above)
Comment (Potherca) [reply milestones.md]: That is correct, the original split for Nr. 1. milestones was rolled back, so the header/title remains untouched (i.e. the way it is now in `master`).

File: milestones.md (hunk 5 above)
Comment (Potherca) [reply milestones.md]: That is correct, the original split for Nr. 1. milestones was rolled back, so the header/title remains untouched (i.e. the way it is now in `master`).

**Review by michielbdejong — APPROVED** [review]
Great!

**Pull request merged**